src/uu/sort/src/sort.rs

```diff
@@ -1952,6 +1977,8 @@ mod tests {
 
     #[test]
     fn test_parse_byte_count() {
+        let s = System::new_all();
+        let total_memory_size = s.total_memory();
```

Comment:
It'd be great if we could test this without getting the actual memory. That might mean some refactoring though.

Reply:
Can it even be done without getting actual memory? If i need to equate percent with value, I need total memory. Or do you mean just checking for correct input and not verifying exact values?

Reply:
You could, for instance, make the function return something like this:
```rust
enum Size {
    Absolute(u64),
    Percentage(u32), // <- so this is just the percentage
}
```
and then check that in the test. Then you can resolve that to the actual size based on the memory outside of the test.

Reply:
Sorry, I didn't understand this. What should i check for? the parse_byte_count function returns usize, Err. if we can only check for whether it was error or some value, it can be done.

Reply:
The idea was to split up parsing the value and resolving the percentage. So it might look something like:
```rust
match parse_size(input)? {
    Size::Absolute(n) => n,
    Size::Percentage(n) => {
        // Get the total memory here
        // and figure out the size
    }
}
```
Then you can test `parse_size` without having access to the total memory by just checking that the right percentage is parsed. Does that make sense?

Reply:
Ok. I will make changes. I don't know if I understood what you want though. What is the reason for test functions to not have memory size call?

Reply:
I have pushed changes which tests for percentage parsing